contracts/test/v0.8/dev/HeartbeatRequester.test.ts

```diff
@@ -0,0 +1,175 @@
+import { getUsers, Personas } from '../../test-helpers/setup'
+import { ethers } from 'hardhat'
+import { Signer } from 'ethers'
+import {
+  MockAuthorizedForwarder,
+  HeartbeatRequester,
+  MockAggregatorProxy,
+  MockOffchainAggregator,
+} from '../../../typechain'
+import { HeartbeatRequester__factory as HeartbeatRequesterFactory } from '../../../typechain/factories/HeartbeatRequester__factory'
+import { MockAuthorizedForwarder__factory as MockAuthorizedForwarderFactory } from '../../../typechain/factories/MockAuthorizedForwarder__factory'
+import { MockAggregatorProxy__factory as MockAggregatorProxyFactory } from '../../../typechain/factories/MockAggregatorProxy__factory'
+import { MockOffchainAggregator__factory as MockOffchainAggregatorFactory } from '../../../typechain/factories/MockOffchainAggregator__factory'
+import { assert, expect } from 'chai'
+
+let personas: Personas
+let owner: Signer
+let caller1: Signer
+let proxy: Signer
+let aggregator: MockOffchainAggregator
+let aggregatorFactory: MockOffchainAggregatorFactory
+let aggregatorProxy: MockAggregatorProxy
+let aggregatorProxyFactory: MockAggregatorProxyFactory
+let requester: HeartbeatRequester
+let requesterFactory: HeartbeatRequesterFactory
+let authorizedForwarder: MockAuthorizedForwarder
+let authorizedForwarderFactory: MockAuthorizedForwarderFactory
+
+describe('HeartbeatRequester', () => {
```

Comment:
aren't we trying to move to foundry for tests going forward?

Reply:
0.8 allows co-existence of foundry and ts tests. going to implement foundry tests later.